Cli-Shared/Program.cs

```diff
@@ -420,10 +421,17 @@ private static BaseAuthentication CreateAuthentication(OperationArguments operat
 
                 case AuthorityType.AzureDirectory:
                     Git.Trace.WriteLine($"authority for '{operationArguments.TargetUri}' is Azure Directory.");
+                    
+                    Configuration.Entry loginHint;
+                    if (operationArguments.GitConfiguration.TryGetEntry(ConfigPrefix, operationArguments.TargetUri, ConfigLoginHintKey, out loginHint)
```

Comment:
Please move this into `LoadOperationArguments` and add a field to `OperationArguments`. The name `LoginHint` is fine, but something more generic like `AuthenticationHint` might be better.

Please make it type `string` and not `ConfigEntry`. 😄